daml-foundations/daml-ghc/daml-stdlib-src/DA/Internal/Template.daml

```diff
@@ -51,11 +51,20 @@ archive c = exercise c Archive
 stakeholder : Template c => c -> [Party]
 stakeholder c = signatory c ++ observer c
 
--- | HIDE pass the @c@ using type applications, otherwise it's ambiguous
+-- | Look up the contract id associated with a given contract key.
+--
+-- The `c` needs to be passed using an explicit type application. For
+-- instance, if you want to look up a contract of template `Account` by its
+-- key `k`, you must call `lookupByKey @Account k`.
 lookupByKey : forall c k. TemplateKey c k => k -> Update (Optional (ContractId c))
 lookupByKey = internalLookupByKey
 
--- | HIDE pass the @c@ using type applications, otherwise it's ambiguous
+-- | Fetch the contract id and contract data associated with a given
+-- contract key.
+--
+-- The `c` needs to be passed using an explicit type application. For
+-- instance, if you want to fetch a contract of template `Account` by its
+-- key `k`, you must call `fetchByKey @Account k`.
```

Comment:
As above - I'd suggest adding something about authorization required. Maybe also the thing about if it fails, that doesn't mean the contract doesn't exist?